scipy/stats/_stats_py.py

```diff
@@ -4771,42 +4815,52 @@ def statistic(x, y):
     # dtype is the data type for the calculations.  This expression ensures
     # that the data type is at least 64 bit floating point.  It might have
     # more precision if the input is, for example, np.longdouble.
-    dtype = type(1.0 + x[0] + y[0])
+    dtype = type(1.0 + x.ravel()[0] + y.ravel()[0])
```

Comment:
Is there a better way to do this?

Reply:
I was in no-think-just-translate mode. We discussed correcting the calculation to respect user-provided dtype.